File: src/main/java/io/sch/historyscan/domain/contexts/analysis/clocrevisions/CodebaseClocRevisions.java
package io.sch.historyscan.domain.contexts.analysis.clocrevisions;

import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.FileSystemTree;
import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.FileInfo;
import io.sch.historyscan.domain.hexagonalarchitecture.DDDAggregate;

import java.util.List;

@DDDAggregate
public record CodebaseClocRevisions(
        FileSystemTree actualFsTree,
        List<FileInfo> ignoredRevisions,
        List<String> extensions) {

    public CodebaseClocRevisions {
        ignoredRevisions = List.copyOf(ignoredRevisions);
        extensions = List.copyOf(extensions);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/analysis/clocrevisions/CodebaseClocRevisions.kt
package io.sch.historyscan.domain.contexts.analysis.clocrevisions

import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.FileInfo
import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.FileSystemTree
import io.sch.historyscan.domain.hexagonalarchitecture.DDDAggregate

@DDDAggregate
class CodebaseClocRevisions(
    val actualFsTree: FileSystemTree, val ignoredRevisions: List<FileInfo>, val extensions: List<String>
)
